### DISCUSSION TITLE
Card as Link

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#305] [Component suggestions]
<img width="720" alt="visuele weergave van de card as link component" src="https://github.com/user-attachments/assets/732d3125-96a9-4890-ab83-fce0296165d8">

We willen graag meer leren over de Card as Link component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1741613889987249)

## Naam

Card as Link

## Doel

Korte sectie die als geheel een link is naar een gerelateerde pagina.

## Gerelateerde componenten

[Case Card](https://github.com/orgs/nl-design-system/discussions/204)

## Onderzoek

[What makes an accessible card? - Rian Rietveld - Youtube](https://youtu.be/AH2NagMoYYE?si=xKEbsCo0TxGwYmlK&t=1683)
In deze video vertelt Rian dat je liever geen `<a>` om alles in de Card moet zetten, omdat een screenreader dan alle inhoud voorleest.
Ze legt ook uit wat je in dit geval kunt doen, zoals het gebruiken van een `<span>`.

[Cards - Inclusive components](https://inclusive-components.design/cards/)
Uitgebreid artikel over het toegankelijk opzetten van een Card component.

[Block Links, Cards, Clickable regions](https://adrianroselli.com/2020/02/block-links-cards-clickable-regions-etc.html)
Adrian Roselli heeft een uitgebreid blogpost geschreven inclusief video's van het gedrag met screenreaders bij verschillende implementaties. Daarnaast geeft hij ook aan dat een gehele card klikbaar maken ervoor zorgt dat je de tekst  niet gemakkelijk kan selecteren.

**Comment by emma-koole** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-cards-news-card--docs)
- [Figma](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=22998-132405&t=88v5vN7USfQal8Ah-4)
- [GitHub](https://github.com/nl-design-system/denhaag/blob/main/components/CardNews/src/index.scss)
---
### Screenshots / Varianten
**Storybook**

Default

![Visuele weergave van de default state Card as Link component van Den Haag.](https://github.com/user-attachments/assets/52e56ae6-f8bd-47b5-8d4f-c4afba6b8601)

Hover
![Visuele weergave van de hover state Card as Link component van Den Haag.](https://github.com/user-attachments/assets/78ae69e8-060d-40d7-99ae-85855a06e59a)

Focus
![Visuele weergave van de focus state Card as Link component van Den Haag](https://github.com/user-attachments/assets/8302f79f-7884-4cbb-be8e-6d2454b1d326)

Website
![Screenshot van de Card as Link component op de denhaag.nl](https://github.com/user-attachments/assets/e5693093-618a-46e2-9488-6a574d575c7d)

---
### Notities
- Richtlijnen beschikbaar (Storybook).
- In Figma zowel mobile als desktop beschikbaar.
- Het wordt zowel nieuwscard als content card genoemd.
- De card as links worden op dit moment gebruik op de website van [denhaag.nl](https://www.denhaag.nl/nl/). Ze worden gebruikt als ingang naar nieuws(waardige content).

**Comment by emma-koole** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-navigation-card--docs)
- [GitHub](https://github.com/Amsterdam/design-system/blob/develop/storybook/src/components/Card/Card.docs.mdx#with-image)
---
### Screenshots / Varianten
**Storybook**

Default
![Visuele weergave van de default state Card as Link component van Amsterdam.](https://github.com/user-attachments/assets/35ddefb0-387c-432e-a259-4d8a711cb736)

With tagline
![Visuele weergave van de Card as Link component van Amsterdam met een tagline.](https://github.com/user-attachments/assets/bf092142-e15f-4baa-90ac-b1d1e18ef7ed)

With image
![Visuele weergave van de Card as Link component van Amsterdam met een afbeelding.](https://github.com/user-attachments/assets/35ecba4f-0908-4735-aeea-abbd4c2768c1)

---
### Notities
- Richtlijnen beschikbaar (Storybook): voor de card component gelden ook de richtlijnen voor links en headings.
- Er is geen border aanwezig.

**Comment by emma-koole** [comment]
## Rotterdam
---
### Screenshots / Varianten

**Website**
![Visuele weergave van de Card as Link component op de publieke website van Rotterdam.](https://github.com/user-attachments/assets/bf13cc1b-03e6-43e5-9d95-9c3147573d2f)

**Comment by emma-koole** [comment]
## RVO

- [Documentatiewebsite](https://nl-design-system.github.io/rvo/docs/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-card--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=2734-118)
- [Github](https://github.com/nl-design-system/rvo/tree/master/components/card)

---
### Screenshots / Varianten
**Storybook**

Standard
![Visuele weergave van de standard Card as Link component van RVO.](https://github.com/user-attachments/assets/9867f995-acb5-4292-b372-57d109ffa91c)

Outline
![Visuele weergave van de Card as Link component met outline van RVO.](https://github.com/user-attachments/assets/0609cd9e-74b4-470a-b79d-3ec974ac98d5)

Colored background
![Visuele weergave van de Card as Link component met een gekleurde achtergrond van RVO.](https://github.com/user-attachments/assets/7e396cc8-aabe-4068-b8b4-1e996132fb21)

Inverted colors
![Visuele weergave van de Card as Link component met omgekeerde kleuren van RVO.](https://github.com/user-attachments/assets/2cf32b91-8c70-4575-be93-f5bab74654ee)

Image (small)
![Visuele weergave van de Card as Link component met een kleine afbeelding van RVO.](https://github.com/user-attachments/assets/4e86e022-535e-4b0b-a524-ba0240702e2c)
mall)

Image (big)
![Visuele weergave van de Card as Link component met een grote afbeelding van RVO.](https://github.com/user-attachments/assets/f693cdb7-6d26-4100-857d-5aab1c344ebe)

Image background
![Visuele weergave van de Card as Link component met een afbeelding als achtergrond van RVO.](https://github.com/user-attachments/assets/72589752-f6bc-43e7-a1b6-646674d11373)

---
### Notities

- Geen hover states aanwezig
- Geen links

**Comment by emma-koole** [comment]
## Onderzoek

[**NN-Group over Cards: UI-Component Definition**](https://www.nngroup.com/articles/cards-component/)
Quite often, cards use a slight drop shadow to show depth, which is a clickability signifier. For most implementations, [clicking or tapping anywhere on the card link to a details page ](https://www.nngroup.com/articles/cards-component/)(though some cards, like the one in our previous example, also include some secondary call-to-action buttons or links in addition to the main link available). This larger touch zone substantially improves usability on both touchscreen devices and mouse-based devices (due to [Fitts’ Law](http://www.asktog.com/columns/022DesignedToGiveFitts.html)).

**Comment by erikkroes** [comment]
Lastige dingen altijd, die cards!

Geabstraheerd is de "card as link" zoiets als een flinke link met veel inhoud (namelijk: de hele card). Waar loop je dan potentieel tegenaan? (Los van de content van de card, die natuurlijk ook goed moet zijn)

- [WCAG 1.3.1](https://www.w3.org/Translations/WCAG22-nl/#info-and-relationships): Afhankelijk van context en toepassing, kan het zijn dat "Informatie, structuur en relaties" overgebracht worden door de presentatie. Zo kan het zijn dat je meerdere cards in een lijst zet, dat je in de code ook duidelijk wil maken dat je een groepering hebt, of dat je card standaard (visueel) een kopje heeft (en die wil je ook in de code).
- [WCAG 1.3.2](https://www.w3.org/Translations/WCAG22-nl/#meaningful-sequence): Let op je hierarchie in code en visueel. Visueel hebben cards vaak een afbeelding als eerste onderdeel. In de code wil je mogelijk de heading als eerste hebben.
- [WCAG 1.4.11](https://www.w3.org/Translations/WCAG22-nl/#non-text-contrast): Een link is een "user interface component". Je wil dat dat wat de link identificeert (herkenbaar maakt) voldoende contrast heeft.
- [WCAG 2.1.1](https://www.w3.org/Translations/WCAG22-nl/#keyboard): Met betrekking tot deze, maar ook algemener. Wat is het interactieve deel, en is dit los te benaderen? ( Dit is -Card as Link-. Dat is 1 groot interactief ding. Willen we ook cards met verschillende interactieve items als content? Dat is een ander patroon, en kan verwarrend zijn als ze beide beschikbaar zijn en gecombineerd worden.)
- [WCAG 2.4.4](https://www.w3.org/Translations/WCAG22-nl/#link-purpose-in-context)/[WCAG 2.5.3](https://www.w3.org/Translations/WCAG22-nl/#label-in-name)/[WCAG 4.1.2](https://www.w3.org/WAI/WCAG22/quickref/?showtechniques=131%2C246%2C253#name-role-value): 3 criteria rond eenzelfde punt; deze belangrijk voor card as link. Welk naam geef je het ding? En hoe verhoudt zich dat tot wat je ziet? En hoe werkt dat bijvoorbeeld met spraaksoftware? Kritiek punt.
- [WCAG 2.4.6](https://www.w3.org/Translations/WCAG22-nl/#headings-and-labels): Je wilt -goede- kopjes voor je cards!

Algemenere kwesties, en mogelijk niet nodig expliciet te noemen:
- [WCAG 1.4.1](https://www.w3.org/Translations/WCAG22-nl/#use-of-color): Ontwerp je card niet zo dat hij op kleur leunt.
- [WCAG 1.4.3](https://www.w3.org/Translations/WCAG22-nl/#contrast-minimum): Vooral gericht op content, er moet genoeg contrast zijn.
- [WCAG 1.4.4](https://www.w3.org/Translations/WCAG22-nl/#resize-text): Wat gebeurt er als de tekst schaalt, past alles dan nog?
- [WCAG 1.4.5](https://www.w3.org/Translations/WCAG22-nl/#images-of-text): Maar al te vaak komen er afbeeldingen met tekst op cards terecht. Mogelijk iets om in de documentatie te noemen.
- [WCAG 2.4.7](https://www.w3.org/Translations/WCAG22-nl/#focus-visible): Je wil het component (zichtbaar) kunnen focussen.
- [WCAG 2.5.8](https://www.w3.org/Translations/WCAG22-nl/#target-size-minimum): Je wil een minimumformaat voor je card. 🤷
- [WCAG 3.2.4](https://www.w3.org/Translations/WCAG22-nl/#consistent-identification): Als je naar pagina's gaat linked wil je consistent zijn in je benaming. Meer een content-ding (en website-breed/component-oversteigend), dan voor het component zelf.


Naast de genoemde resources, ook nog een oude video (met audio problemen): https://www.youtube.com/watch?v=7sE2IsU8X0c

**Comment by emma-koole** [comment]
## DUO
[Website](https://uno.dfront.rijkscloud.nl/#/components/card)

### Screenshots / Varianten
**Geen afbeelding**
![een card zonder afbeelding uit het DUO UNO Rijksoverheid Design System](https://github.com/user-attachments/assets/1d3b0622-5752-43d9-a8ad-de1f014f346e)

**Flag card - met afbeelding**
![een flag card met afbeelding uit het DUO UNO Rijksoverheid Design System](https://github.com/user-attachments/assets/e794ffed-5b7d-409b-8b9d-6faa0ae3f3f3)

**Card met afbeelding**
![een card met afbeelding uit het DUO UNO Rijksoverheid Design System8](https://github.com/user-attachments/assets/c8830c79-f5e7-4027-9be2-793641096327)

**Comment by eslook** [comment]
We hebben dit onderwerp ook besproken in het Developer Open Hour van donderdag 27 maart 2025. Bij deze een samenvatting van wat er besproken is. @mennovanslooten Kan jij deze comment nog even editten met jouw bevindingen?

### Card as Link versus Card zonder Link

Wat als de Card geen Link bevat, maar verder visueel precies (minus de link) een Card is? 

* Maken we dan gebruik van Card as Link component, variant zonder link?
* Word het een los Card component met zijn eigen discussie en eigen documentatie?

### Implementatie van een volledig klikbare Card

Er zijn meerdere manieren om dit op te lossen;

* aria-describedby
* position relative op de card met een position absolute'd pseudo element welke de hele card overlapt
* volledige card in een anchor element
    * In lange lijsten met veel informatie in elke card (metadata, breadcrumb, titel, omschrijving) wil je deze oplossing mogelijk niet, omdat het onoverzichtelijk word
* het gebruik van JavaScript om bij interactie met de card, een link te openen
* een combinatie van een link en JavaScript

Een fancy versie van Jaws (bron: Robbert, edit: [Fusion](https://www.freedomscientific.com/products/software/fusion/)) maakt het mogelijk om te wijzen naar elementen, en daar de informatie van op te lezen. De pseudo selector oplossing kan dit in de weg staan voor bijvoorbeeld informatie over de afbeelding.
Moet een Card uberhaupt wel volledig klikbaar zijn? Is dit een aangeleerd patroon dat mogelijk afgeleerd moet worden?

### Wat is een Card (of een Card as Link)?

Niet elk blokje met een link is een Card as Link.
Bijvoorbeeld voor **Search Result**. Het kan visueel hetzelfde zijn; een volledige klikbare card aldanwel zonder afbeelding. Maar afhankelijk van de zoekresultaten, kan er de wens zijn om verschillende weergaven van de card te hebben.
Stel je hebt **verschillende Cards**; een Search Result Card, Product Card, News Card - schrijf je dan ook dubbele documentatie voor elk component?
In het geval van een **Table Row**, welke visueel hetzelfde kan zijn als een Card as Link:

* Moet de volledige Table Row klikbaar zijn? Dit maakt de tekst minder goed selecteerbaar, vooral in een Table kan dit vervelend zijn.

Hebben we mogelijk een **Simple Card as Link** versus **Complex Card as Link** nodig?
Is een Card mogelijk meer eenzelfde iets als een **Modal Dialog**, **Drawer** of **Accordion**, waar je een aantal functionaliteiten vast hebt staan, maar de body volledig vrij ingevuld kan worden?

**Reply by Robbert** [reply]
Met de fancy versie van Jaws bedoelde ik [Fusion](https://www.freedomscientific.com/products/software/fusion/)

**Comment by emma-koole** [comment]
## ✨ Dit component is nu Help Wanted ✨


Help je mee hem door de Community stappen te krijgen?
[Card as Link op het Community bord](https://github.com/nl-design-system/backlog/issues/101)

**Comment by Ollie-nl** [comment]
Wat bij ons nu online staat, maar een update gaat krijgen en nodig heeft:

![Screen Shot 2025-04-08 at 13 12 19](https://github.com/user-attachments/assets/bcd449f4-b3b6-45ce-a0f0-55723ceeedf7)

![Screen Shot 2025-04-08 at 13 11 17](https://github.com/user-attachments/assets/6eee77cb-cb53-4814-ada4-9c013ebbeabc)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Card as Link op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=card+as+link)

**Comment by Marwaxhello** [comment]
## Sittard-geleen

[website](https://sittard-geleen.nl/)

---

### Screenshots

#### Icon Card
<img width="300" alt="Screenshot 2026-07-13 at 11 32 35" src="https://github.com/user-attachments/assets/9eb81ae6-0303-4e77-8147-78672d411b19" />

#### Image Card
<img width="300" alt="Screenshot 2026-07-13 at 11 33 06" src="https://github.com/user-attachments/assets/22f29e8b-8f10-4e37-8ee0-983a35ade3b0" />

#### News Card
<img width="300" alt="Screenshot 2026-07-13 at 11 34 53" src="https://github.com/user-attachments/assets/b39c38c3-9863-404f-8360-dbb3e5e1f7ae" />

**Comment by jeffreylauwers** [comment]
[GitHub Discussion: Patroon zonder Naam](https://github.com/orgs/nl-design-system/discussions/422) bevat [een visie op Cards](https://github.com/orgs/nl-design-system/discussions/422#discussioncomment-17353679). Zo ook Card as Link.
